File: OnboardingFrontend/src/models/employee.ts
import type { TeamSummary } from './team'

export interface EmployeeSearchResult {
  id: number
  firstName: string
  lastName: string
  jobTitle: string
  teamId: number
  teamName: string
}

export interface EmployeeDetails {
  id: number
  firstName: string
  lastName: string
  email: string
  jobTitle: string
  startDate: string
  currentOnboardingStep: string
  onboardingCompletedAt: string | null
  team: TeamSummary
  manager: ManagerSummary | null
}

export interface ManagerSummary {
  id: number
  firstName: string
  lastName: string
  email: string
  jobTitle: string
}

File: OnboardingBackend/Models/Employee.cs
namespace OnboardingBackend.Models;

public class Employee
{
    public int Id { get; set; }
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public required string Email { get; set; }
    public required string JobTitle { get; set; }
    public DateOnly StartDate { get; set; }

    public int TeamId { get; set; }
    public Team Team { get; set; } = null!;

    public int? ManagerId { get; set; }
    public Employee? Manager { get; set; }
    public ICollection<Employee> DirectReports { get; set; } = [];

    public string CurrentOnboardingStep { get; set; } = "welcome";
    public DateTimeOffset? OnboardingCompletedAt { get; set; }
}
